### PR TITLE
feat: update `postcss-selector-parser`

### PR DESCRIPTION
**Pull request by evilebottnawi**
Problems: https://github.com/postcss/postcss-selector-parser/issues/142#issuecomment-388074112

Problem places in comments.
Some tests will be changes and fixes.
Feel free to feedback.

Btw thanks for great work. Looks we don't have many issues.

/cc @chriseppstein

### CODE REVIEW

File: tests/css_attribute/spaces.css [hunk 1]
```diff
@@ -89,27 +141,43 @@ type
 i
 ])) {}
 [foo|att=val] {}
-[ foo | att = val ] {}
-[  foo  |  att  =  val  ] {}
+[ foo|att = val ] {}
```
Comment (evilebottnawi) [inline tests/css_attribute/spaces.css]: before and after `|` should be no spaces

File: tests/template_literals/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -371,31 +371,31 @@ exports[`styled-jsx-with-expressions.js 1`] = `
...
+   div.\${expr} {
+    color: red;
+   }
+  all\${expr} {
+    color: red;
+  }
   }
   @font-face {
-    \${expr};
+    \${expr}
```
Comment (evilebottnawi) [inline tests/template_literals/__snapshots__/jsfmt.spec.js.snap]: :confused: investigate

File: tests/css_combinator/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -13,7 +13,7 @@ exports[`combinator.css 1`] = `
 .x
     .y {}
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
--Option/root .public/section ~ .public/section:before {
+-option/root .public/section ~ .public/section:before {
```
Comment (evilebottnawi) [inline tests/css_combinator/__snapshots__/jsfmt.spec.js.snap]: I think it is expected

File: tests/css_less/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -2928,7 +2928,8 @@ selector {
   /* ... */
 }
 
-.mixin (@a) when (@a > 10), (@a < -10) {
+.mixin (@a) when (@a > 10),
+(@a < -10) {
```
Comment (evilebottnawi) [inline tests/css_less/__snapshots__/jsfmt.spec.js.snap]: We should switch fox less mixin on value parser, we can do this in other PR

**Comment by chriseppstein** [comment]
postcss/postcss-selector-parser#142 is fixed by https://www.npmjs.com/package/postcss-selector-parser/v/5.0.0-rc.2

**Comment by evilebottnawi** [comment]
@chriseppstein looks we still have problem with nested combinator selectors

```
.parent ~ {
  .child-one {}
  .child-two {}
}
```

On `.parent ~`

```
console.log src/language-css/parser-postcss.js:26
    TypeError: Cannot read property '0' of undefined
        at Parser.combinator (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/parser.js:634:27)
        at Parser.space (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/parser.js:813:18)
        at Parser.parse (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/parser.js:930:22)
        at Parser.loop (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/parser.js:921:18)
        at new Parser (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/parser.js:174:14)
        at Processor._root (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/processor.js:47:22)
        at Processor._runSync (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/processor.js:83:25)
        at Processor.astSync (/home/evilebottnawi/IdeaProjects/prettier/node_modules/postcss-selector-parser/dist/processor.js:121:21)
        at parseSelector (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:23:8)
        at parseNestedCSS (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:303:25)
        at parseNestedCSS (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:229:7)
        at parseNestedCSS (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:229:7)
        at parseWithParser (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:490:24)
        at Object.parse (/home/evilebottnawi/IdeaProjects/prettier/src/language-css/parser-postcss.js:518:12)
        at Object.parse (/home/evilebottnawi/IdeaProjects/prettier/src/main/parser.js:75:19)
        at coreFormat (/home/evilebottnawi/IdeaProjects/prettier/src/main/core.js:69:25)
        at format (/home/evilebottnawi/IdeaProjects/prettier/src/main/core.js:249:18)
        at formatWithCursor (/home/evilebottnawi/IdeaProjects/prettier/src/main/core.js:258:12)
        at Object.formatWithCursor (/home/evilebottnawi/IdeaProjects/prettier/index.js:23:15)
        at prettyprint (/home/evilebottnawi/IdeaProjects/prettier/tests_config/run_spec.js:126:27)
        at fs.readdirSync.forEach.filename (/home/evilebottnawi/IdeaProjects/prettier/tests_config/run_spec.js:48:22)
        at Array.forEach (<anonymous>)
        at run_spec (/home/evilebottnawi/IdeaProjects/prettier/tests_config/run_spec.js:15:27)
        at Object.<anonymous> (/home/evilebottnawi/IdeaProjects/prettier/tests/css_selector_list/jsfmt.spec.js:1:90)
        at Runtime._execModule (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runtime/build/index.js:520:13)
        at Runtime.requireModule (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runtime/build/index.js:332:14)
        at /home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-jasmine2/build/index.js:101:13
        at Generator.next (<anonymous>)
        at step (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-jasmine2/build/index.js:106:583)
        at /home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-jasmine2/build/index.js:106:813
        at new Promise (<anonymous>)
        at /home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-jasmine2/build/index.js:106:494
        at jasmine2 (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-jasmine2/build/index.js:106:68)
        at runTest (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runner/build/run_test.js:98:10)
        at /home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runner/build/index.js:71:68
        at Generator.next (<anonymous>)
        at step (/home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runner/build/index.js:23:376)
        at /home/evilebottnawi/IdeaProjects/prettier/node_modules/jest-runner/build/index.js:23:536
        at process._tickCallback (internal/process/next_tick.js:178:7)
```

**Comment by chriseppstein** [comment]
@evilebottnawi Apologies, I had the fix for that, but somehow lost it by changing branches, I guess. git reflog saved me. `5.0.0.rc.3` is released and includes that fix now.

**Comment by j-f1** [comment]
@evilebottnawi What’s the status on this?

**Comment by evilebottnawi** [comment]
@j-f1 WIP

**Comment by j-f1** [comment]
@evilebottnawi Should we fix the conflicts and merge or should this be closed?

**Comment by evilebottnawi** [comment]
@j-f1 it needs to be done, but now there is absolutely no time

**Comment by j-f1** [comment]
Closing this for now to clean up the PR backlog, but feel free to reopen once you have time to finish it.

**Pull request closed without merging**